docs/overview.rst

```diff
@@ -0,0 +1,116 @@
+.. _overview:
+
+Overview
+========
+
+The serial engine can be used to run sequential workflows.
+A workflow can consist of arbitrary many steps, with each one running
```

Comment:
What about: `A workflow can consist of an arbitrary number of steps. Each of this steps may run in a different computing environment.` ?

Reply:
Actually, I mention the same thing around line ~60 maybe it can be removed from here completely.